kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java

```diff
@@ -183,25 +180,42 @@ public Map<TopicPartition, OffsetAndMetadata> preCommit(Map<TopicPartition, Offs
   private PartitionedTableId getRecordTable(SinkRecord record) {
     String tableName;
     String dataset = config.getString(BigQuerySinkConfig.DEFAULT_DATASET_CONFIG);
-    String[] smtReplacement = record.topic().split(":");
-
-    if (smtReplacement.length == 2) {
-      dataset = smtReplacement[0];
-      tableName = smtReplacement[1];
-    } else if (smtReplacement.length == 1) {
-      tableName = smtReplacement[0];
+    if (topic2TableMap != null) {
+      // TODO what if the map doesn't contain the topic name?
+      // We don't need an explicit call to sanitize here because if the
+      // sanitize flag is true, then the topic2TableMap would already contain
+      // sanitized table names.
+      tableName = topic2TableMap.get(record.topic());
```

Comment:
@sagarrao12 I could not follow above comment. If the map does not contain the topic, then tableName will be null correct? However in that case we should fall back to using the topic name itself as the table name. Not able to figure where that logic is.

Reply:
That's correct. I added a default return value as you suggested.